sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -3610,8 +3610,10 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
         case Some(x) => throw new ParseException(s"SHOW $x FUNCTIONS not supported", ctx)
     }
     val pattern = Option(ctx.pattern).map(string(_))
-    val functionName = Option(ctx.multipartIdentifier).map(visitMultipartIdentifier)
-    ShowFunctionsStatement(userScope, systemScope, pattern, functionName)
+    val functionName = Option(ctx.multipartIdentifier)
```

Comment:
nit: maybe `unresolvedFun`? it's not a name anymore.

Reply:
or keep it `functionName`, and do
```
ShowFunctions(UnresolvedFunc(functionName), userScope, systemScope, pattern)
```

Reply:
Changed it to `unresolvedFunc` since `functionName` in the second suggestion will be `Option[Seq[String]]`.